File: budgetBuddySystem/src/main/java/edu/miu/cs489/budgetbuddysystem/service/IncomeService.java
package edu.miu.cs489.budgetbuddysystem.service;

import edu.miu.cs489.budgetbuddysystem.model.Income;
import jakarta.persistence.EntityNotFoundException;

import java.util.List;

public interface IncomeService {
    Income addNewIncome(Long userId, Income income);
    List<Income> getAllIncomes(Long userId);
    Income getIncomeById(Long userId, Long incomeId) throws EntityNotFoundException;
    Income updateIncomeById(Long userId, Long incomeId, Income updatedIncome) throws EntityNotFoundException;
    void deleteIncomeById(Long userId, Long incomeId);
}

File: budget-buddy-web/src/services/IncomeService.js
import http from '../config/axios'
import header from "../config/header";
import {INCOME_URL, INCOMES_URL} from "../contants/api";


const getAllIncomes = (userId, accessToken) => {
    return http.get(INCOMES_URL(userId), {headers: header(accessToken)});
}

const addNewIncome = (userId, accessToken, newIncome) => {
    return http.post(INCOMES_URL(userId), newIncome, {headers: header(accessToken)});
}

const getIncomeById = (userId, accessToken, incomeId) => {
    return http.get(INCOME_URL(userId, incomeId), {headers: header(accessToken)})
}

const updateIncomeById = (userId, accessToken, incomeId, updatedIncome) => {
    return http.put(INCOME_URL(userId, incomeId), updatedIncome, {headers: header(accessToken)})
}

const deleteIncomeById = (userId, accessToken, incomeId) => {
    return http.delete(INCOME_URL(userId, incomeId), {headers: header(accessToken)})
}

const IncomeService = {
    getAllIncomes,
    getIncomeById,
    addNewIncome,
    updateIncomeById,
    deleteIncomeById
}

export default IncomeService;